### PR TITLE
Datasets

### PR DESCRIPTION
**Pull request by clee-ai**
I remade the PR to be based off of a branch of this repo.

### CODE REVIEW

File: conf/model/default.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# By default we turn off recursive instantiation, allowing the user to instantiate themselves at the appropriate times.
+_recursive_: false
+
+_target_: torch_points3d.model.base_model.PointCloudBaseModel
```
Comment (humanpose1) [inline conf/model/default.yaml]: models instead of model

File: conf/dataset/segmentation/s3dis/s3dis1x1.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# @package dataset
+defaults:
+  - segmentation/default
+_target_: torch_points3d.dataset.s3dis1x1.s3dis_data_module
```
Comment (humanpose1) [inline conf/dataset/segmentation/s3dis/s3dis1x1.yaml]: datasets instead of dataset

File: torch_points3d/core/instantiator.py [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
 from omegaconf import DictConfig
 
-from lightning_transformers.core import TransformerDataModule
-from lightning_transformers.core.data import TokenizerDataModule
+from torch_points3d.dataset.base_dataset import PointCloudDataModule
 
 if TYPE_CHECKING:
     # avoid circular imports
-    from lightning_transformers.core import TaskTransformer
+    from torch_points3d.model.base_model import PointCloudBaseModel
```
Comment (humanpose1) [inline torch_points3d/core/instantiator.py]: same as above

File: torch_points3d/trainer.py [hunk 4]
```diff
@@ -2,38 +2,42 @@
...
 from pytorch_lightning import LightningDataModule
 from pytorch_lightning.utilities.distributed import rank_zero_info
-import hydra.utils.instantiate as hydra_instantiate
 
-from torch_points3d.datasets.dataset_factory import instantiate_dataset, convert_to_lightning_data_module
-from torch_points3d.datasets.dataset_factory import instantiate_dataset
+
+# from hydra.utils.instantiate as hydra_instantiate
+
+from torch_points3d.model.base_model import PointCloudBaseModel
```
Comment (humanpose1) [inline torch_points3d/trainer.py]: same as above for model and dataset

**Review by humanpose1 — COMMENTED** [review]
It's a detail but we should have the same convention for models/model. 
I propose to keep the same as p3d for the organisation of directories (because there are many datasets and models)
torch_points3d/models
torch_points3d/datasets
As for the conf
we can keep model and dataset because we select one model and one dataset. What do you think ?

INow I have this error
```
TypeError: Error instantiating 'torch_points3d.model.base_model.PointCloudBaseModel' : __init__() missing 1 required positional argument: 'model'
```

But it is because we need to instantiate APIModel (I've merged the branch).
We don't have to solve it in this PR, we can solve it in a future one

**Comment by clee-ai** [comment]
I'm okay with keeping that convention. I'll change that and merge.

As far as the error goes, we can fix the model/dataset later, I just wanted to get the dataset loaded properly.

**Pull request merged**